ceph.spec.in

```diff
@@ -450,13 +450,50 @@ Requires:       python%{_python_buildid}-Werkzeug
 Requires:       python%{_python_buildid}-pyOpenSSL
 Requires:       python%{_python_buildid}-bcrypt
 Recommends:     python%{_python_buildid}-influxdb
+Recommends:	ceph-mgr-diskprediction-local = %{_epoch_prefix}%{version}-%{release}
+Recommends:	ceph-mgr-diskprediction-cloud = %{_epoch_prefix}%{version}-%{release}
+Recommends:	ceph-mgr-rook = %{_epoch_prefix}%{version}-%{release}
 %endif
 %description mgr
 ceph-mgr enables python modules that provide services (such as the REST
 module derived from Calamari) and expose CLI hooks.  ceph-mgr gathers
 the cluster maps, the daemon metadata, and performance counters, and
 exposes all these to the python modules.
 
+%package mgr-diskprediction-local
+Summary:        ceph-mgr diskprediction_local plugin
+%if 0%{?suse_version}
+Group:          System/Filesystems
+%endif
+Requires:       ceph-base = %{_epoch_prefix}%{version}-%{release}
+Requires:       python%{_python_buildid}-numpy
+Requires:       python%{_python_buildid}-scipy
+Requires:       python%{_python_buildid}-scikit-learn
+%description mgr-diskprediction-local
+ceph-mgr-diskprediction-local is a ceph-mgr plugin that tries to predict
+disk failures using local algorithms and machine-learning databases.
+
+%package mgr-diskprediction-cloud
+Summary:        ceph-mgr diskprediction_cloud plugin
+%if 0%{?suse_version}
+Group:          System/Filesystems
+%endif
+Requires:       ceph-base = %{_epoch_prefix}%{version}-%{release}
+%description mgr-diskprediction-cloud
+ceph-mgr-diskprediction-cloud is a ceph-mgr plugin that tries to predict
+disk failures using services in the Google cloud.
+
+%package mgr-rook
+Summary:        ceph-mgr rook plugin
+%if 0%{?suse_version}
+Group:          System/Filesystems
+%endif
+Requires:       ceph-base = %{_epoch_prefix}%{version}-%{release}
+Requires:       python%{_python_buildid}-kubernetes
```

Comment:
RH-specific: Note that this package is currently in epel-testing, so we might hit failing builds on centos7 if this is merged before the package is promoted to the main epel repo. We can accelerate that process by upvoting it here, but you probably don't want to do that without testing it yourself first:

https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2019-2a28713541

Reply:
AIUI it's required at install time, not build time, but yes, it will fail to install if python-kubernetes isn't available, it's true.  The same is currently true of ceph-mgr-diskprediction-local (because its Requires are not in the package repos).

However: 1) python-mgr-rook didn't build (since I don't have a Recommends to install it, my "install ceph" test didn't notice), and I need to find out why, and
2) I think I may have the names of the packages wrong; there is a "numpy" available.  I thought I had been careful here.

Reply:
Ahh yes, you're quite correct. The build shouldn't fail but it will fail at install time. Just be forewarned that we probably can't install this package reliably until it makes it out of epel-testing.

Reply:
should we comment out that line until the epel piece is sorted out?

Reply:
well, this makes it not install for the right reasons, right?

Reply:
removed requirements until dependent packages can be corralled

Reply:
Dependent package (python-kubernetes) should now be available in EPEL7, so I think we can add this back in now.